src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -172,19 +212,27 @@ private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, I
 
                     if (name == null)
                     {
-                        RefKind refMode = !argumentRefKinds.IsDefaultOrEmpty ? argumentRefKinds[argumentIndex] : RefKind.None;
-                        return
-                            refMode == RefKind.None
-                            ? (argumentIndex >= parameters.Length - 1 &&
-                               parameters.Length > 0 &&
-                               parameters[parameters.Length - 1].IsParams &&
-                               // An argument that is an array of the appropriate type is not a params argument.
-                               (boundArguments.Length > argumentIndex + 1 ||
-                                argument.Type.TypeKind != TypeKind.Array ||
-                                !argument.Type.Equals(parameters[parameters.Length - 1].Type, true))
-                               ? (IArgument)new Argument(ArgumentKind.ParamArray, parameters[parameters.Length - 1], CreateParamArray(parameters[parameters.Length - 1], boundArguments, argumentIndex))
-                               : new SimpleArgument(parameters[parameterIndex], argument))
-                            : (IArgument)new Argument(ArgumentKind.Positional, parameters[parameterIndex], argument);
+                        RefKind refMode = argumentRefKinds.IsDefaultOrEmpty ? RefKind.None : argumentRefKinds[argumentIndex];
+
+                        if (refMode != RefKind.None)
+                        {
+                            return (IArgument)new Argument(ArgumentKind.Positional, parameters[parameterIndex], argument);
+                        }
+
+                        if (argumentIndex >= parameters.Length - 1 &&
+                            parameters.Length > 0 &&
+                            parameters[parameters.Length - 1].IsParams &&
+                            // An argument that is an array of the appropriate type is not a params argument.
+                            (boundArguments.Length > argumentIndex + 1 ||
+                             argument.Type.TypeKind != TypeKind.Array ||
+                             !argument.Type.Equals(parameters[parameters.Length - 1].Type, true)))
```

Comment:
BTW, does the type have to match exactly? Can it be an array of derived type for example. I would feel more comfortable if this code was reusing/sharing code with OverloadResolution/LocalRewriter rather than having its own implementation, which is likely to deviate in many non-trivial scenarios.